api/v1beta1/alert_types.go

```diff
@@ -54,6 +54,11 @@ type AlertSpec struct {
 	// Defaults to false.
 	// +optional
 	Suspend bool `json:"suspend,omitempty"`
+
+	// Extra metadata to be added to the alert
+	// +kubebuilder:validation:Optional
+	// +optional
+	ExtraMetadata map[string]string `json:"extraMetadata,omitempty"`
```

Comment:
Please remove this field, the `summary` can be used to set the cluster name, region, etc.

Reply:
By adding this map you get a lot more flexibility around being able to process the alert once it reached the provider. 

The summary as a string means more complex parsing logic at the provider end. There are many tags that may/may not make sense in a summary field which adding a map would allow.

It also better fits in with the Prometheus "labels" paradigm, which many people running Kubernetes would be used to. I'm thinking an "alertmanager" provider could also make heavy use of this in the same way I am using this with the Prometheus provider too.

Reply:
Once we defines the set of annotations for the Event, you could define routing rules based on e.g. `GIT_URL` and check if the URL matches some application repo. Besides the Event annotations, we could include the Alert annotations too, so instead of adding some info in the Alert `.spec.extraMetadata` you would add it to `.metadata.annotations`.

Reply:
For now please map the `summary` field to the opsgenie details.